src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -167,9 +171,16 @@ public function decode(string $data, string $format, array $context = []): mixed
                     foreach ($cols as $col) {
                         $header = explode($keySeparator, $col);
                         $headers[] = $header;
+                        $headerFile[] = $col;
                         $headerCount[] = \count($header);
                     }
 
+                    if (($context[self::CHECK_VALID_HEADERS] ?? $this->defaultContext[self::CHECK_VALID_HEADERS])
+                        && \count(array_intersect($headerFile, $expectedHeader)) !== \count($expectedHeader)
```

Comment:
Isn't `\count(array_intersect($cols, $expectedHeader))` enough?

Moreover, maybe we can firstly check the the length of `$cols` (or `$headerFile`) and `$expectedHeader` are the same? So we avoid an unnecessary resource consuming `array_intersect`

Reply:
Just wondering, is the headers' order matter?

Reply:
I added count checking before array_intersect.
For the sort, I think it doesn't matter because the normalizer doesn't need specific order.
The goal of this PR is just to verify processed fields in the serializer and don't accept other fields. 
What do you think ?

Reply:
LGTM yes!